docs/compression_algorithms/Pruning.md

```diff
@@ -83,6 +84,30 @@ Where ![a, k, b](https://microsoft.codecogs.com/gif.latex?a%2C%20k%2C%20b) - par
 
     // A list of model control flow graph node scopes to be considered for this operation - functions as a 'whitelist'. Optional.
     // "target_scopes": []
-}```
+}
+```
 
 > **NOTE:**  In all our pruning experiments we used SGD optimizer.
+
+
+#### Export pruning algorithm description:
+This algorithm transforms trained with `Filter pruning algorithm` model with masked pruning (when convolution size remains
+ same but some filters are zeroed by mask) to smaller model with actually pruned operations.
+
+The algorithm consists of 3 stages:
+1. On the first stage, channel-wise masks from pruned operations propagate through the graph. Every operation has a function that can calculate the output mask by its attributes and the input masks. These calculated masks determine how the operations will be pruned.
+ There are three groups of operations:
+ - Operations that just pass the mask further unchanged. These operations include all activations and elementwise operations with one input, Batch Norms, Pullings.
+ - Operations that cannot work with pruned input and do not skip the mask further. These operations include reshapes, matmul, reduce with some operations.
```

Comment:
Looks strange, especially the end of the sentence. Do you mean `Reshape, MatMul, ReduceMean`?